src/dispatching/dialogue/storage/mod.rs

```diff
@@ -73,3 +73,85 @@ pub trait Storage<D> {
         chat_id: i64,
     ) -> BoxFuture<'static, Result<Option<D>, Self::Error>>;
 }
+
+/// A storage with an erased error type.
+pub type ErasedStorage<D> = Arc<dyn Storage<D, Error = Box<dyn std::error::Error>>>;
+
+/// Extension methods for working with [`Storage`].
+pub trait StorageExt<D> {
+    /// Returns [`ErasedStorage`].
+    fn erase(self: Arc<Self>) -> ErasedStorage<D>;
```

Comment:
can this method be a part of the `Storage` trait? (with a default implementation)

Reply:
Nope, because of the generics in the implementation ("the `Storage` trait cannot be made into an object").

Reply:
?

It seems to work for me: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=17b6779007212b731a3a99783301421b. Am I missing something?